internal/ethereum/blocklistener.go

```diff
@@ -140,6 +144,28 @@ func (bl *blockListener) listenLoop() {
 		update := &ffcapi.BlockHashEvent{GapPotential: gapPotential}
 		var notifyPos *list.Element
 		for _, h := range blockHashes {
+			if len(h) != 32 {
+				if !bl.allowNonStandardBlockHashLength {
+					log.L(bl.ctx).Errorf("Tried to index a non-standard size block hash length: %s", h.String())
+					failCount++
+					continue
+				}
+
+				if bl.nonStandardBlockHashSizeResolutionMethod == "" {
+					log.L(bl.ctx).Warnf("No non-standard hash length resolution method was provided, defaulting to truncation")
+					bl.nonStandardBlockHashSizeResolutionMethod = "truncate"
+				}
+
+				if bl.nonStandardBlockHashSizeResolutionMethod == "truncate" {
+					h, err = h.Truncate(32)
```

Comment:
```suggestion
h = ethtypes.HexBytes0xPrefix(h[0:32])
```
... combined with above suggestion

Reply:
Yep agreed here - if we go with this approach, we'll do this and then remove the requirement on the firefly-signer PR.